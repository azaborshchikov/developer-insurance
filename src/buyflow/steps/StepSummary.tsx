import React from 'react';
import { Link } from 'react-router-dom';
import { observer } from 'mobx-react-lite';
import { StepComponent, Store } from '../types';

interface IStepSummary extends Store {
  firstName?: string,
  lastName?: string,
  email: string,
  age: number
}

const StepSummary: StepComponent<IStepSummary> = ({ store }) => {
  return <>
    {store.firstName && (
      <div>
        First Name: {store.firstName}
      </div>
    )}
    {store.lastName && (
      <div>
        Last Name: {store.lastName}
      </div>
    )}
    <div>
      Email: {store.email}
    </div>
    <div>
      Age: {store.age}
    </div>
    <div>
      <Link to="/?purchased=dev_ins">
        Purchase
      </Link>
    </div>
  </>;
};

export default observer(StepSummary);
