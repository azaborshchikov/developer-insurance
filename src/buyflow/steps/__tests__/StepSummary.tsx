import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import StepSummary from '../StepSummary';

const FIRST_NAME = 'John';
const LAST_NAME = 'Doe';
const EMAIL = 'a@b.c';
const AGE = 25;

const handleNext = jest.fn();
const store = {
  firstName: FIRST_NAME,
  lastName: LAST_NAME,
  email: EMAIL,
  age: AGE
};

const setup = () => {
  const utils = render(
    <BrowserRouter>
      <StepSummary onNext={handleNext} store={store}/>
    </BrowserRouter>
  );
  return {
    link: utils.getByRole('link'),
    ...utils
  }
};

describe('checks summary step', () => {
  test('render', () => {
    const { container, link } = setup();
    expect(link).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
