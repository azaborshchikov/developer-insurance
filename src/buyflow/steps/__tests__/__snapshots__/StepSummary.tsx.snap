// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks summary step render 1`] = `
<div>
  <div>
    First Name: 
    John
  </div>
  <div>
    Last Name: 
    Doe
  </div>
  <div>
    Email: 
    a@b.c
  </div>
  <div>
    Age: 
    25
  </div>
  <div>
    <a
      href="/?purchased=dev_ins"
    >
      Purchase
    </a>
  </div>
</div>
`;
